os/kernel/binary_manager/binary_manager_getinfo.c

```diff
@@ -276,19 +203,12 @@ void binary_manager_get_info_all(int requester_pid)
 #ifdef CONFIG_APP_BINARY_SEPARATION
 	/* User binaries data */
 	bin_count = binary_manager_get_ucount();
-	if (bin_count > 0) {
-		for (bin_idx = 1; bin_idx <= bin_count; bin_idx++) {
-			size = binary_manager_get_available_size(bin_idx);
-			if (size < 0) {
-				response_msg.result = BINMGR_OPERATION_FAIL;
-				break;
-			}
-			response_msg.result = BINMGR_OK;
-			response_msg.data.bin_info[result_idx].available_size = size;
-			strncpy(response_msg.data.bin_info[result_idx].name, BIN_NAME(bin_idx) , BIN_NAME_MAX);
-			response_msg.data.bin_info[result_idx].version = (double)BIN_LOADVER(bin_idx);
-			result_idx++;
-		}
+	for (bin_idx = 1; bin_idx <= bin_count; bin_idx++) {
+		response_msg.result = BINMGR_OK;
+		response_msg.data.bin_info[result_idx].available_size = BIN_PARTSIZE(bin_idx, (BIN_USEIDX(bin_idx) ^ 1));
+		strncpy(response_msg.data.bin_info[result_idx].name, BIN_NAME(bin_idx) , BIN_NAME_MAX);
+		response_msg.data.bin_info[result_idx].version = BIN_VER(bin_idx, BIN_USEIDX(bin_idx));
+		result_idx++;
 	}
 
 	if (response_msg.result == BINMGR_OK) {
```

Comment:
`response_msg.result` is always OK, because of initialization in line 190.
Please check this.